File: checkers/197/main.cpp
#include <checkers/testlib.h>
#include <iostream>
#include <vector>
#include <set>
#include <algorithm>
#include <map>
#include <set>
#include <string>
#include <sstream>

using namespace NTestlib;

int f[8] = {-2, -2, -1, -1, 1, 1, 2, 2};
int s[8] = {1, -1, 2, -2, -2, 2, -1, 1};

int main(int argc, char *argv[])
{
  InitChecker(argc, argv);
  int n = File.ReadInt(), m = File.ReadInt();
  std::vector<std::vector<int>> c(n + 1, std::vector<int>(m + 1, 0));
  int temp = Out.ReadInt();
  if (temp == -1)
    if (n == 4 && m == 4)
      QuitWith(AC, "" Full solution "");
    else
      QuitWith(WA, "" Wrong answer "");
  if (temp < 1 || temp > n)
  {
    QuitWith(PE, "" Index out of range "");
  }
  int x = temp, y = Out.ReadInt(1, m);
  c[x][y] = 1;
  int x2 = x, y2 = y;
  for (int i = 0; i < n * m - 1; i++)
  {
    bool isFind = false;
    int x1 = Out.ReadInt(1, n), y1 = Out.ReadInt(1, m);
    for (int j = 0; j < 8; j++)
    {
      if (x1 - x == f[j] && y1 - y == s[j])
      {
        isFind = true;
        break;
      }
    }
    if (!isFind)
    {
      std::stringstream msg;
      msg << "" WA Incorrect step : x = "" << x1 << "";
      y = "" << y1;
      QuitWith(WA, msg.str());
    }
    if (c[x1][y1] == 1 && !(x1 == x2 && y1 == y2))
      QuitWith(WA, "" Twice "");
    c[x1][y1] = 1;
    x = x1;
    y = y1;
  }
  QuitWith(AC, "" Full solution "");
}

File: checkers/197/main.go
package main

import (
	"fmt"
	"testlib"
	"testlib/core"
)

func i2b(i_0 int) bool {
	if i_0 == 0 {
		return false
	} else {
		return true
	}
}

func createMatrix0(n int, m int) [][]int {
	matrix := make([][]int, n)
	for i := 0; i < n; i++ {
		matrix[i] = make([]int, m)
		for j := 0; j < m; j++ {
			matrix[i][j] = 0
		}
	}
	return matrix
}

func checkValueOutOfSegmen(val int, a int, b int) bool {
	if val < a || val > b {
		return true
	} else {
		return false
	}
}

func main() {
	defer func() {
		_ = testlib.GetStream(testlib.TaskOutput).Close()
		_ = testlib.GetStream(testlib.InputFromUser).Close()
	}()
	inputStream := testlib.GetReader(testlib.TaskInput)
	//outputStream := testlib.GetReader(testlib.TaskOutput)
	inputFromUserStream := testlib.GetReader(testlib.InputFromUser)

	f := []int{-2, -2, -1, -1, 1, 1, 2, 2}
	s := []int{1, -1, 2, -2, -2, 2, -1, 1}
	n, m := core.ReadInt(inputStream), core.ReadInt(inputStream)
	c := createMatrix0(n+1, m+1)
	temp := core.ReadInt(inputFromUserStream)
	if temp == -1 {
		if n == 4 && m == 4 {
			testlib.WriteAnswer(fmt.Sprintf("OK"), testlib.OK)
			return
		} else {
			testlib.WriteAnswer(fmt.Sprintf("WA"), testlib.WA)
			return
		}
	}
	if temp < 1 || temp > n {
		testlib.WriteAnswer(fmt.Sprintf("PE Index out of range"), testlib.PE)
		return
	}
	x, y := temp, core.ReadInt(inputFromUserStream)
	if y < 1 || y > m {
		testlib.WriteAnswer(fmt.Sprintf("PE Index out of range"), testlib.PE)
		return
	}
	c[x][y] = 1
	x2, y2 := x, y
	for i := 0; i < n*m-1; i++ {
		isFind := false
		x1, y1 := core.ReadInt(inputFromUserStream), core.ReadInt(inputFromUserStream)
		if y1 < 1 || y1 > m || x1 < 1 || x1 > n {
			testlib.WriteAnswer(fmt.Sprintf("PE Index out of range"), testlib.PE)
			return
		}
		for j := 0; j < 8; j++ {
			if x1-x == f[j] && y1-y == s[j] {
				isFind = true
				break
			}
		}
		if !isFind {
			testlib.WriteAnswer(fmt.Sprintf("WA Incorrect step : x = %d, y = %d", x1, y1), testlib.WA)
			return
		}
		if c[x1][y1] == 1 && !(x1 == x2 && y1 == y2) {
			testlib.WriteAnswer(fmt.Sprintf("WA Twice"), testlib.WA)
			return
		}
		c[x1][y1] = 1
		x = x1
		y = y1
	}
	testlib.WriteAnswer(fmt.Sprintf("Full solution"), testlib.OK)
}
